### PR TITLE
doc: document config-manager plugin

### LINKED ISSUE #1282: Documentation for config-manager
- Document command config-manager
- Document that config-manager modification of repository configuration uses configuration overrides and does not modify the original repo file
- Document that config-manager modification of configuration modify dnf.conf and not other overrides

This is a follow-up to #1011.

### PR DESCRIPTION
**Pull request by jrohel**
Closes #1282

### CODE REVIEW

File: doc/dnf5_plugins/config-manager.8.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      Repository ID may contain globs. In this case, the `repoid` pattern is resolved and override is removed from all matching sections.
+
+``setvar [--create-missing-dir] <variable_name=value>+``
+    | Sets one or more variables.
+
+      Variables are loaded from multiple directories. The list of directory paths is taken from the `varsdir` option.
+      The `setvar` command stores variables to the last directory in the list (by default "/etc/dnf/vars"). Variables from this directory are read last and override the values of the same variables defined in previous directories. In other words, the last directory has the highest priority.
+
+      Note:
+      The variables "releasever_major" and "releasever_minor" are read-only. Their values are generated from the "releasever" variable.
```
Comment (evan-goode) [inline doc/dnf5_plugins/config-manager.8.rst:86]: Good thought to include this bit about read-only variables.

File: doc/dnf5_plugins/config-manager.8.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+``--set=REPO_OPTION=VALUE``
+    | Set option in newly created repository.
+
+
+Examples
+========
+
+``dnf5 config-manager addrepo --from-repofile=http://example.com/some/additional.repo``
+    | Download "additional.repo", test it and put it in repository configuration directory.
```
Comment (evan-goode) [inline doc/dnf5_plugins/config-manager.8.rst:126]: Add comma: `test it, and put it`

File: doc/dnf5_plugins/config-manager.8.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+``dnf5 config-manager addrepo --set=baseurl=http://example.com/different/rep --id=example --set=enabled=0``
+    | Create new repo file with "http://example.com/different/repo" as `baseurl`. Set repository ID to "example" and disable it.
+
+``dnf5 config-manager setopt repoid1.enabled=1 repoid2.enabled=0``
+    | Sets override to enable repository identified by "repoid1" and disable repository identified by "repoid2".
+
+``dnf5 config-manager setopt repo1.proxy=http://proxy.example.com:3128/ repo2.proxy=http://proxy.example.com:3128/``
+    | Sets override for `proxy` option in repositories with repository IDs "repo1" and "repo2".
+
+``dnf5 config-manager setopt *-debuginfo.gpgcheck=0``
```
Comment (evan-goode) [inline doc/dnf5_plugins/config-manager.8.rst:140]: Maybe put single quotes around these wildcards, or else the shell would expand them.

**Review by evan-goode — CHANGES_REQUESTED** [review]
Overall I think this is clear and well-structured. There's a lot of detail about the behavior of each subcommand and how the different overrides files interact, but I think it's warranted. The structure of our configuration is complex, and users will undoubtedly run into some edge case with config-manager and turn to this doc page to find out what's going on.

Something big that's missing is documentation of the changes from the DNF 4 config-manager plugin. The CLI is completely different: https://dnf-plugins-core.readthedocs.io/en/latest/config_manager.html, I'm sure for good reason, but when reading this page, the first goal of many users will be to figure out how to update their DNF 4 scripts.

This could be done in a separate PR, but I would like to see a section added to https://dnf5.readthedocs.io/en/latest/changes.html about config-manager that has examples of DNF 4 commands and their (rough?) DNF 5 equivalents.

File: doc/dnf5.8.rst [hunk 4]
```diff
@@ -421,12 +421,11 @@ Application Plugins:
     | :manpage:`dnf5-automatic(8)`, :ref:`Automatic command <automatic_plugin_ref-label>`
     | :manpage:`dnf5-builddep(8)`, :ref:`Builddep command <builddep_plugin_ref-label>`
     | :manpage:`dnf5-changelog(8)`, :ref:`Changelog command <changelog_plugin_ref-label>`
+    | :manpage:`dnf5-config-manager(8)`, :ref:`Config-manager command <config_manager_plugin_ref-label>`
```
[the remark this answers is not in the document]
Comment (jrohel) [reply doc/dnf5.8.rst:424]: OK. I'll leave it as it is. We should unify the documentation in another PR.

**Comment by jrohel** [comment]
@evan-goode Thank you for your review.

**Review by evan-goode — APPROVED** [review]
Great, thanks!

**Pull request merged**